### ISSUE TITLE
'cat_name' columns have incompatible types

### ISSUE THREAD

**Issue by stscieisenhamer** [spacetelescope/jwst#2938]
# Issue
From D-string test run of the WholeWorks regression on 2018-12-05 produces the following error from `calwebb_image3`:

<details><summary>Show log</summary>
<p>

```
strun cfgs/calwebb_image3.cfg jw82600-o011_20181205t224433_image3_002_asn.json
2018-12-12 14:53:29,078 - stpipe.Image3Pipeline - INFO - Image3Pipeline instance created.
2018-12-12 14:53:29,080 - stpipe.Image3Pipeline.tweakreg - INFO - TweakRegStep instance created.
2018-12-12 14:53:29,081 - stpipe.Image3Pipeline.skymatch - INFO - SkyMatchStep instance created.
2018-12-12 14:53:29,082 - stpipe.Image3Pipeline.outlier_detection - INFO - OutlierDetectionStep instance created.
2018-12-12 14:53:29,084 - stpipe.Image3Pipeline.resample - INFO - ResampleStep instance created.
2018-12-12 14:53:29,085 - stpipe.Image3Pipeline.source_catalog - INFO - SourceCatalogStep instance created.
2018-12-12 14:53:29,086 - stpipe - INFO - Hostname: inanna2.local.stsci.edu
2018-12-12 14:53:29,086 - stpipe - INFO - OS: Darwin
2018-12-12 14:53:29,121 - stpipe.Image3Pipeline - INFO - Step Image3Pipeline running with args ('jw82600-o011_20181205t224433_image3_002_asn.json',).
2018-12-12 14:53:29,668 - stpipe.Image3Pipeline - INFO - Prefetching reference files for dataset: 'jw82600011001_02103_00001_nrcb4_cal.fits' reftypes = ['drizpars']
2018-12-12 14:53:30,917 - stpipe.Image3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/Users/eisenham/Downloads/crds-cache/references/jwst/nircam/jwst_nircam_drizpars_0001.fits'.
2018-12-12 14:53:31,241 - stpipe.Image3Pipeline - INFO - Prefetching reference files for dataset: 'jw82600011001_02103_00001_nrcb2_cal.fits' reftypes = ['drizpars']
2018-12-12 14:53:31,244 - stpipe.Image3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/Users/eisenham/Downloads/crds-cache/references/jwst/nircam/jwst_nircam_drizpars_0001.fits'.
2018-12-12 14:53:31,569 - stpipe.Image3Pipeline - INFO - Prefetching reference files for dataset: 'jw82600011001_02103_00001_nrcb3_cal.fits' reftypes = ['drizpars']
2018-12-12 14:53:31,572 - stpipe.Image3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/Users/eisenham/Downloads/crds-cache/references/jwst/nircam/jwst_nircam_drizpars_0001.fits'.
2018-12-12 14:53:31,946 - stpipe.Image3Pipeline - INFO - Prefetching reference files for dataset: 'jw82600011001_02103_00001_nrcb1_cal.fits' reftypes = ['drizpars']
2018-12-12 14:53:31,949 - stpipe.Image3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/Users/eisenham/Downloads/crds-cache/references/jwst/nircam/jwst_nircam_drizpars_0001.fits'.
2018-12-12 14:53:32,289 - stpipe.Image3Pipeline - INFO - Prefetching reference files for dataset: 'jw82600011001_02104_00001_nrcb4_cal.fits' reftypes = ['drizpars']
2018-12-12 14:53:32,292 - stpipe.Image3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/Users/eisenham/Downloads/crds-cache/references/jwst/nircam/jwst_nircam_drizpars_0001.fits'.
2018-12-12 14:53:32,648 - stpipe.Image3Pipeline - INFO - Prefetching reference files for dataset: 'jw82600011001_02104_00001_nrcb1_cal.fits' reftypes = ['drizpars']
2018-12-12 14:53:32,651 - stpipe.Image3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/Users/eisenham/Downloads/crds-cache/references/jwst/nircam/jwst_nircam_drizpars_0001.fits'.
2018-12-12 14:53:33,011 - stpipe.Image3Pipeline - INFO - Prefetching reference files for dataset: 'jw82600011001_02104_00001_nrcb2_cal.fits' reftypes = ['drizpars']
2018-12-12 14:53:33,013 - stpipe.Image3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/Users/eisenham/Downloads/crds-cache/references/jwst/nircam/jwst_nircam_drizpars_0001.fits'.
2018-12-12 14:53:33,441 - stpipe.Image3Pipeline - INFO - Prefetching reference files for dataset: 'jw82600011001_02104_00001_nrcb3_cal.fits' reftypes = ['drizpars']
2018-12-12 14:53:33,444 - stpipe.Image3Pipeline - INFO - Prefetch for DRIZPARS reference file is '/Users/eisenham/Downloads/crds-cache/references/jwst/nircam/jwst_nircam_drizpars_0001.fits'.
2018-12-12 14:53:33,444 - stpipe.Image3Pipeline - INFO - Starting calwebb_image3 ...
2018-12-12 14:53:36,563 - stpipe.Image3Pipeline - INFO - Aligning input images...
2018-12-12 14:53:36,634 - stpipe.Image3Pipeline.tweakreg - INFO - Step tweakreg running with args (<ModelContainer>,).
2018-12-12 14:53:36,877 - stpipe.Image3Pipeline.tweakreg - INFO - Detected 2 sources in jw82600011001_02103_00001_nrcb4_cal.fits.
2018-12-12 14:53:37,154 - stpipe.Image3Pipeline.tweakreg - INFO - Detected 17 sources in jw82600011001_02103_00001_nrcb2_cal.fits.
2018-12-12 14:53:37,490 - stpipe.Image3Pipeline.tweakreg - INFO - Detected 28 sources in jw82600011001_02103_00001_nrcb3_cal.fits.
2018-12-12 14:53:37,762 - stpipe.Image3Pipeline.tweakreg - INFO - Detected 16 sources in jw82600011001_02103_00001_nrcb1_cal.fits.
2018-12-12 14:53:37,936 - stpipe.Image3Pipeline.tweakreg - INFO - Detected 1 sources in jw82600011001_02104_00001_nrcb4_cal.fits.
2018-12-12 14:53:38,118 - stpipe.Image3Pipeline.tweakreg - WARNING - /Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/photutils/detection/findstars.py:977: AstropyUserWarning: Sources were found, but none pass the sharpness and roundness criteria.
  'and roundness criteria.', AstropyUserWarning)

2018-12-12 14:53:38,122 - stpipe.Image3Pipeline.tweakreg - WARNING - No sources found in jw82600011001_02104_00001_nrcb1_cal.fits.
2018-12-12 14:53:38,287 - stpipe.Image3Pipeline.tweakreg - WARNING - /Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/photutils/detection/findstars.py:977: AstropyUserWarning: Sources were found, but none pass the sharpness and roundness criteria.
  'and roundness criteria.', AstropyUserWarning)

2018-12-12 14:53:38,289 - stpipe.Image3Pipeline.tweakreg - WARNING - No sources found in jw82600011001_02104_00001_nrcb2_cal.fits.
2018-12-12 14:53:38,500 - stpipe.Image3Pipeline.tweakreg - WARNING - /Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/photutils/detection/findstars.py:977: AstropyUserWarning: Sources were found, but none pass the sharpness and roundness criteria.
  'and roundness criteria.', AstropyUserWarning)

2018-12-12 14:53:38,506 - stpipe.Image3Pipeline.tweakreg - WARNING - No sources found in jw82600011001_02104_00001_nrcb3_cal.fits.
----------------------------------------------------------------------
ERROR RUNNING STEP 'Image3Pipeline':
    The 'cat_name' columns have incompatible types: ['str1280',
    'float64', 'float64', 'float64']
Traceback (most recent call last):
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/astropy/table/operations.py", line 853, in _vstack
    out[out_name] = col_cls.info.new_like(cols, n_rows, metadata_conflicts, out_name)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/astropy/table/column.py", line 173, in new_like
    ('meta', 'unit', 'format', 'description'))
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/astropy/utils/data_info.py", line 643, in merge_cols_attributes
    out['dtype'] = metadata.common_dtype(cols)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/astropy/utils/metadata.py", line 64, in common_dtype
    raise tme
astropy.utils.metadata.MergeConflictError: Arrays have incompatible types ['str1280', 'float64', 'float64', 'float64']

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/jwst-0.13.0b0.dev44+gca84f0a7-py3.7-macosx-10.7-x86_64.egg/jwst/stpipe/cmdline.py", line 297, in step_from_cmdline
    step.run(*positional)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/jwst-0.13.0b0.dev44+gca84f0a7-py3.7-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/jwst-0.13.0b0.dev44+gca84f0a7-py3.7-macosx-10.7-x86_64.egg/jwst/pipeline/calwebb_image3.py", line 70, in process
    input_models = self.tweakreg(input_models)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/jwst-0.13.0b0.dev44+gca84f0a7-py3.7-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/jwst-0.13.0b0.dev44+gca84f0a7-py3.7-macosx-10.7-x86_64.egg/jwst/tweakreg/tweakreg_step.py", line 127, in process
    wgroup = WCSGroupCatalog(wcsimlist, name=wcsimlist[0].name)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/jwst-0.13.0b0.dev44+gca84f0a7-py3.7-macosx-10.7-x86_64.egg/jwst/tweakreg/wcsimage.py", line 736, in __init__
    self._catalog = self.create_group_catalog()
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/jwst-0.13.0b0.dev44+gca84f0a7-py3.7-macosx-10.7-x86_64.egg/jwst/tweakreg/wcsimage.py", line 867, in create_group_catalog
    return table.vstack(catalogs, join_type='exact')
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/astropy/table/operations.py", line 292, in vstack
    out = _vstack(tables, join_type, col_name_map, metadata_conflicts)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/astropy/table/operations.py", line 858, in _vstack
    .format(out_name, err._incompat_types))
astropy.table.np_utils.TableMergeError: The 'cat_name' columns have incompatible types: ['str1280', 'float64', 'float64', 'float64']

----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/astropy/table/operations.py", line 853, in _vstack
    out[out_name] = col_cls.info.new_like(cols, n_rows, metadata_conflicts, out_name)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/astropy/table/column.py", line 173, in new_like
    ('meta', 'unit', 'format', 'description'))
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/astropy/utils/data_info.py", line 643, in merge_cols_attributes
    out['dtype'] = metadata.common_dtype(cols)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/astropy/utils/metadata.py", line 64, in common_dtype
    raise tme
astropy.utils.metadata.MergeConflictError: Arrays have incompatible types ['str1280', 'float64', 'float64', 'float64']

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/jwst-0.13.0b0.dev44+gca84f0a7-py3.7-macosx-10.7-x86_64.egg/EGG-INFO/scripts/strun", line 26, in <module>
    step = Step.from_cmdline(sys.argv[1:])
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/jwst-0.13.0b0.dev44+gca84f0a7-py3.7-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 162, in from_cmdline
    return cmdline.step_from_cmdline(args)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/jwst-0.13.0b0.dev44+gca84f0a7-py3.7-macosx-10.7-x86_64.egg/jwst/stpipe/cmdline.py", line 297, in step_from_cmdline
    step.run(*positional)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/jwst-0.13.0b0.dev44+gca84f0a7-py3.7-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/jwst-0.13.0b0.dev44+gca84f0a7-py3.7-macosx-10.7-x86_64.egg/jwst/pipeline/calwebb_image3.py", line 70, in process
    input_models = self.tweakreg(input_models)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/jwst-0.13.0b0.dev44+gca84f0a7-py3.7-macosx-10.7-x86_64.egg/jwst/stpipe/step.py", line 400, in run
    step_result = self.process(*args)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/jwst-0.13.0b0.dev44+gca84f0a7-py3.7-macosx-10.7-x86_64.egg/jwst/tweakreg/tweakreg_step.py", line 127, in process
    wgroup = WCSGroupCatalog(wcsimlist, name=wcsimlist[0].name)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/jwst-0.13.0b0.dev44+gca84f0a7-py3.7-macosx-10.7-x86_64.egg/jwst/tweakreg/wcsimage.py", line 736, in __init__
    self._catalog = self.create_group_catalog()
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/jwst-0.13.0b0.dev44+gca84f0a7-py3.7-macosx-10.7-x86_64.egg/jwst/tweakreg/wcsimage.py", line 867, in create_group_catalog
    return table.vstack(catalogs, join_type='exact')
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/astropy/table/operations.py", line 292, in vstack
    out = _vstack(tables, join_type, col_name_map, metadata_conflicts)
  File "/Users/eisenham/anaconda3/envs/condadev/lib/python3.7/site-packages/astropy/table/operations.py", line 858, in _vstack
    .format(out_name, err._incompat_types))
astropy.table.np_utils.TableMergeError: The 'cat_name' columns have incompatible types: ['str1280', 'float64', 'float64', 'float64']
```

</p>
</details>

**Comment by stscieisenhamer** [comment]
# Resources
Data is located in
```
/grp/jwst/ssb/eisenhamer/jwst_issues/issue2938
```

**Comment by hbushouse** [comment]
Could this be due to the fact that 3 of the exposures had zero sources found and hence their source catalogs may be either non-existent or incorrect format (due to being empty)?

**Comment by mcara** [comment]
I doubt it is because of catalog being empty. This sounds like a bug due to catalog name being assigned different _types_ of variables. I need to think about how to fix this in a nice way (i.e., do I even want to merge metadata?)

**Comment by mcara** [comment]
@hbushouse 

> Could this be due to the fact that 3 of the exposures had zero sources found and hence their source catalogs may be either non-existent or incorrect format (due to being empty)?

Yep, you are 100% correct... => I was wrong :)

**Comment by mcara** [comment]
Should be fixed via #2939 once it is merged.

**Comment by mcara** [comment]
#2939 was just merged. It should be safe to close this issue.
